cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftService.java

```diff
@@ -0,0 +1,141 @@
+      this.metrics.gauge(Constants.Metrics.DRAFT_COUNT, store.getDraftCount());
+    } catch (TransactionException e) {
+      LOG.warn("Failed to write {} metric on DraftService startup due to: {}", Constants.Metrics.DRAFT_COUNT,
+               e.getMessage());
+    }
+  }
+
+  /**
+   * Returns a sorted and filtered list of drafts for the given namespace and owner
+   *
+   * @param namespaceSummary the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param includeConfig the returned Draft objects will not include the pipeline config if this is false
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required. Note
+   *   that the sortField matches against the column names in the table spec found in {@link DraftStore}
+   * @param filter string used to do case-insensitive prefix matching on the draft name
+   * @return List of {@link Draft}
+   * @throws TransactionException when an error occurs while fetching from the table
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespaceSummary, String owner,
+                                boolean includeConfig, SortRequest sortRequest,
+                                @Nullable String filter) throws TransactionException {
+    List<Draft> drafts = store.listDrafts(namespaceSummary, owner, sortRequest);
+    if (!includeConfig) {
+      drafts.forEach(DraftRequest::clearConfig);
+    }
+
+    if (!Strings.isNullOrEmpty(filter)) {
+      //TODO Add filtering on other fields?
+      drafts = drafts.stream()
+        .filter(draft -> draft.getName().toLowerCase().startsWith(filter.toLowerCase()))
+        .collect(Collectors.toList());
+    }
+    return drafts;
+
+  }
+
+  /**
+   * Fetch the given draft
+   *
+   * @param draftId {@link DraftId} that is used to uniquely identify a draft
+   * @return the {@link Draft} object
+   * @throws TransactionException when an error occurs while fetching from the table
+   * @throws DraftNotFoundException if the draft does not exist
+   */
+  public Draft getDraft(DraftId draftId) throws TransactionException, DraftNotFoundException {
+    return store.getDraft(draftId).orElseThrow(() -> new DraftNotFoundException(draftId));
+  }
+
+  /**
+   * Write the given draft
+   *
+   * @param draftId {@link DraftId} that is used to uniquely identify a draft
+   * @param draftRequest {@link DraftRequest} that contains the rest of the draft data
+   * @throws TransactionException when an error occurs while writing to the table
+   */
+  public <T extends ETLConfig> void writeDraft(DraftId draftId,
+                                               DraftRequest<T> draftRequest) throws TransactionException {
+    //TODO add collision detection using the hashes
+    try {
+      store.writeDraft(draftId, draftRequest);
+    } catch (TransactionException e) {
+      LOG.warn("Failed to write {} metric on writeDraft request due to: {}", Constants.Metrics.DRAFT_COUNT,
```

Comment:
So, just log without failing? Is this the expected behavior?

Reply:
Also a side effect of testing, this has been fixed